scipy/sparse/_coo.py

```diff
@@ -330,10 +356,15 @@ def toarray(self, order=None, out=None):
         fortran = int(B.flags.f_contiguous)
         if not fortran and not B.flags.c_contiguous:
             raise ValueError("Output array must be C or F contiguous")
-        M,N = self.shape
+        if self.ndim > 2:
+            raise ValueError("Cannot densify higher-rank sparse array")
+        M, N, *_ = self.shape + (1, 1)
```

Comment:
Why do we need to do this?

Reply:
This is a bit of a cryptic trick to handle 1d or 0d cases: by concatenating the shape tuple with two extra 1s, we can assign M and N regardless of the original shape.

Reply:
I see. I think we should at least document that:
```suggestion
        # This handles both 0D and 1D cases correctly regardless of the original shape.
        M, N, *_ = self.shape + (1, 1)
```

Alternatively, we could just explicitly set `M` and `N` for both cases.